packages/wallet-ts/src/utils/wallets/welldone/WelldoneWallet.ts

```diff
@@ -0,0 +1,213 @@
+import { WalletProvider } from './welldone'
+
+const $window = (typeof window !== 'undefined' ? window : {}) as Window & {
+  dapp?: WalletProvider
+}
+
+export class WelldoneWallet {
+  private chainId: CosmosChainId | TestnetCosmosChainId | ChainId
+
+  private chainName: string
+
+  constructor(chainId: CosmosChainId | TestnetCosmosChainId | ChainId) {
+    this.chainId = chainId
+    this.chainName = chainId.split('-')[0]
+  }
+
+  async getAccounts(): Promise<{ address: string; pubKey: string }> {
+    const welldone = await this.getWelldoneWallet()
+
+    try {
+      const accounts = await welldone.request(this.chainName, {
+        method: 'dapp:accounts',
+      })
+
+      if (Object.keys(accounts).length === 0) {
+        throw new Error(
+          `Please make the ${this.chainName} account in the WELLDONE wallet`,
+        )
+      }
+
+      return accounts[this.chainName]
+    } catch (e: unknown) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        contextModule: WalletErrorActionModule.GetAccounts,
+      })
+    }
+  }
+
+  public async broadcastTx(txRaw: CosmosTxV1Beta1Tx.TxRaw): Promise<string> {
+    const welldone = await this.getWelldoneWallet()
+
+    try {
+      const { hash } = await welldone.request(this.chainName, {
+        method: 'dapp:sendSignedTransaction',
+        params: [
+          `0x${Buffer.from(
+            CosmosTxV1Beta1Tx.TxRaw.encode(txRaw).finish(),
+          ).toString('hex')}`,
+        ],
+      })
+
+      return hash
+    } catch (e) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        context: 'WELLDONE',
+        contextModule: 'bradcast-tx',
+      })
+    }
+  }
+
+  public async waitTxBroadcasted(txHash: string): Promise<TxResponse> {
+    const endpoints = await this.getChainEndpoints()
+
+    return new TxRestApi(endpoints.rest).fetchTxPoll(txHash)
+  }
+
+  private changePubKeyType(authInfoBytes: Uint8Array) {
+    const authInfo = CosmosTxV1Beta1Tx.AuthInfo.decode(authInfoBytes)
+
+    authInfo.signerInfos.forEach((signerInfo) => {
+      if (signerInfo.publicKey) {
+        signerInfo.publicKey.typeUrl = '/cosmos.crypto.secp256k1.PubKey'
+      }
+    })
+
+    return CosmosTxV1Beta1Tx.AuthInfo.encode(authInfo).finish()
+  }
+
+  public async signTransaction(
+    signDoc: Omit<CosmosTxV1Beta1Tx.SignDoc, 'accountNumber'> & {
+      accountNumber: Long
+    },
+  ): Promise<DirectSignResponse> {
+    const authInfoBytes = this.changePubKeyType(signDoc.authInfoBytes)
+    signDoc.authInfoBytes = authInfoBytes
+
+    const welldone = await this.getWelldoneWallet()
+
+    try {
+      const signBytes = makeSignBytes(signDoc)
+      const response = await welldone.request(this.chainName, {
+        method: 'dapp:signTransaction',
+        params: [`0x${Buffer.from(signBytes).toString('hex')}`],
+      })
+
+      return {
+        signed: signDoc,
+        signature: {
+          pub_key: encodeSecp256k1Pubkey(
```

Comment:
Have you tested this on Injective tho? Because we are using another pubkey encoding/signing scheme and I think this will not work for us.

Reply:
@bangjelkoski 
When I tested it with [simple counter example](https://github.com/InjectiveLabs/injective-simple-sc-counter-ui) it worked well. But I agree that it's wrong. I'll fix it.

Reply:
okay, thanks, ping me when its done!

Reply:
https://github.com/InjectiveLabs/injective-ts/blob/06e2f8565b342fa10c488bb3adbf9fc2668e4e78/packages/sdk-ts/src/core/modules/tx/utils/tx.ts#L14-L46 you can use this function

Reply:
@bangjelkoski Thank you! I modified the code using `getPublicKey` function. Please check it.